### PR TITLE
Add a bridge to identify the currently selected tab name

### PR DESCRIPTION
**Pull request by pepopowitz**
In #3389, I demonstrated a proof of concept on how I'd go about identifying the currently selected tab name from a React component. This PR cleans up that implementation, and is limited to setting up the bridge.

A followup PR will take advantage of this new bridge function to modify user flow within the sell tab landing page.

### CODE REVIEW

File: src/lib/NativeModules/TopMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { NativeModules } from "react-native"
+
+const { ARTopMenuModule } = NativeModules
+
+export function getSelectedTabName() {
```
Comment (pepopowitz) [inline src/lib/NativeModules/TopMenu.tsx:5]: I'm not totally sure about this extra layer of abstraction. I was following the pattern of the `ARSwitchBoardModule`, but as this is really just a pass-through, it feels unnecessary. I could also see it becoming a custom hook as I work through the follow-up PR that's dedicated to my specific navigation issue.

File: Artsy/View_Controllers/App_Navigation/ARTopMenuNavigationDataSource.m [hunk 2]
```diff
@@ -217,6 +219,27 @@ - (NSUInteger)indexForTabType:(ARTopTabControllerTabType)tabType
     return [self.tabOrder indexOfObject:@(tabType)];
 }
 
+- (NSString *)tabNameForIndex:(NSInteger)index
+{
+    ARTopTabControllerTabType tab = [self tabTypeForIndex:index];
```
Comment (pepopowitz) [inline Artsy/View_Controllers/App_Navigation/ARTopMenuNavigationDataSource.m:224]: @pvinis I think you mentioned that you had a trick for emitting the name of an enum based on its value. I didn't want to spend a lot of time here, so I hardcoded it; happy to update if someone's got a better approach.

File: src/lib/NativeModules/TopMenu.tsx (hunk 1 above)
Comment (pepopowitz) [reply src/lib/NativeModules/TopMenu.tsx:5]: And yeah, [that's how it turned out](https://github.com/artsy/eigen/pull/3402/files#diff-1a95080af2a04563f97f830b674e87b5) - it holds my custom hook.

**Comment by ds300** [comment]
talked to @pepopowitz on slack and we decided to refactor this as follow-up to unblock other things :+1:

**Pull request merged**